### ISSUE TITLE
[dataio] files should support the `with` statement in python (Trac #1499)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1491]
We've been using python 2.6+ for a while now, so let's support the `with` statement for I3Files:

```text
with dataio.I3File("foo.i3") as f:
   frame = f.pop_frame()
```

Migrated from https://code.icecube.wisc.edu/ticket/1499
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T21:50:46", 
    "description": "We've been using python 2.6+ for a while now, so let's support the `with` statement for I3Files:\n\n{{{\nwith dataio.I3File(\"foo.i3\") as f:\n   frame = f.pop_frame()\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547243446066124", 
    "component": "combo core", 
    "summary": "[dataio] files should support the `with` statement in python", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-01-04T20:46:50", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160104T21:26:34: 
* david.schultz changed status from "new" to "closed"
* david.schultz changed resolution from "" to "fixed"
* david.schultz commented:

> In changeset 2376:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2376"
> the with statement for I3File works.  closes #1491
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:15: 
* olivas commented:

> Milestone February 2016 deleted


* olivas changed milestone from "February 2016" to ""

**Comment by icecube-trac** [comment]
Trac update at 20190111T21:50:46: nwhitehorn commented:

> In changeset 169180:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="169180"
> the with statement for I3File works.  closes #1491
> ```
